sdk/core/azure-core/src/main/java/com/azure/core/util/polling/LocationPollingStrategy.java

```diff
@@ -0,0 +1,179 @@
+                if (retryAfter != null) {
+                    return new PollResponse<>(status, pollResponse,
+                        Duration.ofSeconds(Long.parseLong(retryAfter)));
+                } else {
+                    return new PollResponse<>(status, pollResponse);
+                }
+            });
+        });
+    }
+
+    @SuppressWarnings("unchecked")
+    @Override
+    public Mono<U> getResult(PollingContext<T> pollingContext, TypeReference<U> resultType) {
+        if (pollingContext.getLatestResponse().getStatus() == LongRunningOperationStatus.FAILED) {
+            return Mono.error(new RuntimeException("Long running operation failed."));
+        } else if (pollingContext.getLatestResponse().getStatus() == LongRunningOperationStatus.USER_CANCELLED) {
+            return Mono.error(new RuntimeException("Long running operation canceled."));
+        }
+
+        String finalGetUrl;
+        String httpMethod = pollingContext.getData(HTTP_METHOD);
+        if ("PUT".equalsIgnoreCase(httpMethod) || "PATCH".equalsIgnoreCase(httpMethod)) {
+            finalGetUrl = pollingContext.getData(REQUEST_URL);
+        } else if ("POST".equalsIgnoreCase(httpMethod) && pollingContext.getData(LOCATION) != null) {
+            finalGetUrl = pollingContext.getData(LOCATION);
+        } else {
+            throw logger.logExceptionAsError(new RuntimeException("Cannot get final result"));
+        }
+
+        if (finalGetUrl == null) {
+            String latestResponseBody = pollingContext.getData(POLL_RESPONSE_BODY);
+            if (TypeUtil.isTypeOrSubTypeOf(BinaryData.class, resultType.getJavaType())) {
+                return (Mono<U>) Mono.just(BinaryData.fromString(latestResponseBody));
+            } else {
+                return Mono.fromCallable(() -> serializer.deserialize(latestResponseBody, resultType.getJavaType(),
+                    SerializerEncoding.JSON));
+            }
+        } else {
+            HttpRequest request = new HttpRequest(HttpMethod.GET, finalGetUrl);
+            Mono<HttpResponse> responseMono;
+            if (context == null) {
+                responseMono = httpPipeline.send(request);
+            } else {
+                responseMono = httpPipeline.send(request, context);
+            }
+            return responseMono.flatMap(res -> {
+                if (TypeUtil.isTypeOrSubTypeOf(BinaryData.class, resultType.getJavaType())) {
+                    return (Mono<U>) BinaryData.fromFlux(res.getBody());
+                } else {
+                    return res.getBodyAsString().flatMap(body -> Mono.fromCallable(() ->
+                        serializer.deserialize(body, resultType.getJavaType(), SerializerEncoding.JSON)));
+                }
+            });
+        }
+    }
+
+    @Override
+    public Mono<T> cancel(PollingContext<T> pollingContext, PollResponse<T> initialResponse) {
+        return Mono.error(new IllegalStateException("Cancellation is not supported."));
+    }
```

Comment:
Is this true for all services that use LocationPollingStrategy? Is it worth including a cancellation lambda in the constructor like PollerFlux has?

Reply:
Cancellation support is case-by-case - meaning that there's no guideline on how cancellation API should be designed. Therefore there's no generic cancellation implementation we can do in azure-core.

Right now if an API supports cancellation, the SDK author must extend this polling strategy and override the `cancel()` method. It's possible to add another constructor overload to support passing in a cancellation method supplier.